### DISCUSSION TITLE
Related or Calculated Field

### DISCUSSION THREAD

**Discussion by biapar** [leekelleher/umbraco-contentment#329] [Idea]
Calculate Field: that is a field with a value that is related with a values of other fields.

**Comment by leekelleher** [comment]
@biapar Good suggestion, thank you.

It is kind of already possible using either the [Templated Label](https://github.com/leekelleher/umbraco-contentment/blob/develop/docs/editors/templated-label.md) or [Render Macro](https://github.com/leekelleher/umbraco-contentment/blob/develop/docs/editors/render-macro.md) labels.

Render Macro uses a C#/Razor partial-view, which has access to `@Model.Content`, so can get the property values, etc.

Templated Label is an AngularJS template, so done client-side, and would require a bit of hacking about with the `$scope` to access the other properties, e.g. `$parent.$parent.node.variants[0].tabs[0].properties[0].value` ... ugly, but doable.

I have recently added support for `vm.currentPage` (see commit: f0208c91d6559c927f1bb1b953252c789567309b) in the Templated Label. which replaces the `$parent.$parent.node` bit, but still it'll put your AngularJS skills to the test. I'd probably go with the Macro route myself.

...or did you have something else in mind?

**Reply by biapar** [reply]
Thanks to your reply. 
I think about a field where its value is the result of a math operation with the values of other fields

**Comment by leekelleher** [comment]
I'd noticed this post on the Our Umbraco forum, cross-linking for reference:
https://our.umbraco.com/forum/using-umbraco-and-getting-started/112167-computed-label-based-on-document-type-properties

**Reply by biapar** [reply]
But I don't understand where is the computed field...

**Reply by leekelleher** [reply]
I've no idea. The replier on the forum post either has their own custom editor for this, or is making it up. 🤷
